go-controller/pkg/ovn/ovn.go

```diff
@@ -504,6 +506,17 @@ func (oc *Controller) WatchNodes() error {
 	_, err := oc.watchFactory.AddNodeHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			node := obj.(*kapi.Node)
+			noHostSubnet, err := oc.noHostSubnet(node)
+			if err != nil {
+				logrus.Errorf("error with ignored node label: %v", err)
+			}
+			if noHostSubnet {
+				oc.lsMutex.Lock()
+				defer oc.lsMutex.Unlock()
+				oc.logicalSwitchCache[node.Name] = nil
```

Comment:
@JacobTanenbaum let's also add the bit to pods.go that checks `if val, ok := oc.logicalSwitchCache[node.Name]; ok && val == nil {` and then does an early return from addLogicalPort().

Reply:
fixed